shared/output/cloud/compute_small_wrapper/lib/google/cloud/compute.rb

```diff
@@ -0,0 +1,225 @@
+# frozen_string_literal: true
+
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Auto-generated by gapic-generator-ruby. DO NOT EDIT!
+
+# Require this file early so that the version constant gets defined before
+# requiring "google/cloud". This is because google-cloud-core will load the
+# entrypoint (gem name) file, which in turn re-requires this file (hence
+# causing a require cycle) unless the version constant is already defined.
+require "google/cloud/compute/version"
+
+require "googleauth"
+gem "google-cloud-core"
+require "google/cloud" unless defined? ::Google::Cloud.new
+require "google/cloud/config"
+
+# Set the default configuration
+::Google::Cloud.configure.add_config! :compute do |config|
+  config.add_field! :endpoint,      "compute.googleapis.com", match: ::String
+  config.add_field! :credentials,   nil, match: [::String, ::Hash, ::Google::Auth::Credentials]
+  config.add_field! :scope,         nil, match: [::Array, ::String]
+  config.add_field! :lib_name,      nil, match: ::String
+  config.add_field! :lib_version,   nil, match: ::String
+  config.add_field! :timeout,       nil, match: ::Numeric
+  config.add_field! :metadata,      nil, match: ::Hash
+end
+
+module Google
+  module Cloud
+    module Compute
+      ##
+      # Create a new client object for Addresses.
+      #
+      # By default, this returns an instance of
+      # [Google::Cloud::Compute::V1::Addresses::Rest::Client](https://googleapis.dev/ruby/google-cloud-compute-v1/latest/Google/Cloud/Compute/V1/Addresses/Rest/Client.html)
+      # for version V1 of the API.
+      # However, you can specify specify a different API version by passing it in the
+      # `version` parameter. If the Addresses service is
+      # supported by that API version, and the corresponding gem is available, the
+      # appropriate versioned client will be returned.
+      #
+      # ## About Addresses
+      #
+      # Services
+      #
+      # The Addresses API.
+      #
+      # @param version [::String, ::Symbol] The API version to connect to. Optional.
+      #   Defaults to `:v1`.
+      # @return [Addresses::Rest::Client] A client object for the specified version.
+      #
+      def self.addresses version: :v1, &block
```

Comment:
With combined Rest-and-Grpc libraries still an eventual goal, should there be a transport selector, similar to the version selector?

Reply:
Yes, there will be, for libraries where we do have both transports.